### DISCUSSION TITLE
Relative assets

### DISCUSSION THREAD

**Discussion by davestewart** [ElMassimo/iles#123] [Q&A] [answered]
In VuePress you can locate images locally, then reference using a relative path:

```
 +- content
     +- blog
         +- some-post
             +- index.md
             +- assets
                 +- thumb.png
                 +- image.png
```

In `/content/blog/some-post/index.md`:

```md
---
title: Some post
thumbnail: ./assets/thumb.png
---

My cool post!

<img src="./assets/image.png" />

...
```

When the site is previewed or built, everything just works.

Does îles do the same?

**Comment by ElMassimo** [comment, marked as answer]
Hi Dave! @davestewart 

The second case should [work out of the box](https://github.com/ElMassimo/iles/blob/main/packages/mdx/src/remark-mdx-images.ts#L26-L27) in îles.

For the frontmatter asset reference, see [this discussion](https://github.com/ElMassimo/iles/discussions/82).

**Reply by davestewart** [reply]
Thanks! I do something similar in my project to run correction on images from a social plugin.

It would be cool if this was somehow a first class citizen.

Using `./` is the most common way.

It would be great if îles' support for front matter did that!

**Reply by ElMassimo** [reply]
I haven't found the need myself, because the images I would reference from the frontmatter, I'm already transforming using `@islands/images` and [applying a preset](https://github.com/ElMassimo/iles/discussions/82#discussioncomment-2092564).

It's a tricky one, because for that to work reliably îles would need to inject an import statement that Vite can detect, but `frontmatter` is plain data and needs to be serializable.

Open to explorations here, perhaps there's a way to resolve to a "built" path using Vite's internal APIs, pull requests are welcome!
